### PR TITLE
Destination redshift: improve memory usage?

### PR DESCRIPTION
**Pull request by edgao**
* Pass `Stream<PartialAirbyteMessage>` instead of List
* go to one thread instead of two

@gisripa + my theory is that every thread effectively duplicates the data in a batch of records (by rendering it into a giant SQL blob). So going down to one thread should reduce the memory pressure we're seeing.

Sticking with a stream instead of list probably won't be a _huge_ impact since the underlying "big data" elements are passed by reference, but it does reduce the number of object references we're dealing with and is a better interface anyway.

Largely similar to https://github.com/airbytehq/airbyte/pull/33070, but based on master instead of the dv2 branch.

prerelease publish running https://github.com/airbytehq/airbyte/actions/runs/7093320413; I'll test this in the dv1 perf test connection https://cloud.airbyte.com/workspaces/b61bc266-ef3c-460c-af2f-f70da4a2993c/connections/3a93c4ec-9b61-42f8-b6f2-4ec052129dfc/job-history

### CODE REVIEW

File: airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/JdbcBufferedConsumerFactory.java [hunk 1]
```diff
@@ -80,7 +80,7 @@ public static SerializedAirbyteMessageConsumer createAsync(final Consumer<Airbyt
         catalog,
         new BufferManager((long) (Runtime.getRuntime().maxMemory() * 0.2)),
         defaultNamespace,
-        Executors.newFixedThreadPool(2));
+        Executors.newFixedThreadPool(1));
```
Comment (gisripa) [inline airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/JdbcBufferedConsumerFactory.java:83]: How do you feel about using 10% memory factor and 2 threads ?

**Comment by gisripa** [comment]
I'm curious if there is heap improvement by adding `-XX:+UseStringDeduplication` to jvm options.

File: airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/JdbcBufferedConsumerFactory.java (hunk 1 above)
Comment (edgao) [reply airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/JdbcBufferedConsumerFactory.java:83]: :shrug: I'll kick off a sync with the current branch and let it run overnight and we can reevaluate tomorrow morning I guess

Comment (edgao) [reply airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/JdbcBufferedConsumerFactory.java:83]: first attempt (with 0.2x factor + 1 thread) still got killed. Second attempt succeeded in 3 hours

I'll try 0.1x factor I guess :/ but I think that means we can't reliably go to 2 threads

Comment (edgao) [reply airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/JdbcBufferedConsumerFactory.java:83]: ... and I guess I can try enabling the string dedup jvm option later, too 🤔 need to find where to actually inject the jvm flags though

**Comment by edgao** [comment]
watching jstat run during the sync (1 thread, 0.1x multiplier). Memory usage does in fact sometimes spike when a flush is triggered (old space nearly full, eden at 250%, survivor space at 2000% :joy:)
image

old capacity is 460276KB + eden capacity is 184256KB, so that puts us at >900MB peak utilization... sync hasn't crashed yet but it's pretty close to the limit. I don't think we can reliably push it any further without giving it more resources.

**Comment by edgao** [comment]
sync still getting killed on its first attempt :(

**Comment by edgao** [comment]
sync eventually succeeded but took nearly 5 hours :(

**Pull request closed without merging**